Marlin/src/lcd/menu/menu_advanced.cpp

```diff
@@ -366,7 +366,7 @@ void menu_backlash();
           MPC_EDIT_ITEMS(e);
           END_MENU();
         };
-        #define MPC_ENTRY(N) SUBMENU_N(N, MSG_MPC_EDIT, []{ mpc_edit_hotend(MenuItemBase::itemIndex); });
+        #define MPC_ENTRY(N) SUBMENU_N(N, MSG_MPC_EDIT, [&]{ mpc_edit_hotend(MenuItemBase::itemIndex); });
```

Comment:
The only change needed is to use the same default capture for the outer lambda as the inner lambda.

Reply:
I tried that first, along with line 354, doesn't compile:

`Marlin\src\lcd\menu\menu_item.h:275:40: error: cannot convert 'menu_advanced_temperature()::<lambda()>' to 'screenFunc_t' {aka 'void (*)()'}`